docs/reference/asyncapi-rules.md

```diff
@@ -0,0 +1,311 @@
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+asyncapi: "2.0.0"
+info:
+  title: Awesome API
+  description: A very well defined API
+  version: "1.0"
+  contact:
+    name: A-Team
+    email: a-team@goarmy.com
+    url: https://goarmy.com/apis/support
+```
+
+### asyncapi2-info-contact
+
+Info object should contain `contact` object.
+
+Hopefully your API description document is so good that nobody ever needs to contact you with questions, but that is rarely the case. The contact object has a few different options for contact details.
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+asyncapi: "2.0.0"
+info:
+  title: Awesome API
+  version: "1.0"
+  contact:
+    name: A-Team
+    email: a-team@goarmy.com
+```
+
+### asyncapi2-info-description
+
+OpenAPI object info `description` must be present and non-empty string.
+
+Examples can contain Markdown so you can really go to town with them, implementing getting started information like where to find authentication keys, and how to use them.
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+asyncapi: "2.0.0"
+info:
+  version: "1.0.0"
+  title: Descriptive API
+  description: >+
+    Some description about the general point of this API, and why it exists when another similar but different API also exists.
+
+    ## Authentication
+
+    This API uses OAuth2 and tokens can be requested from [Dev Portal: Tokens](https://example.org/developers/tokens).
+```
+
+### asyncapi2-info-license-url
+
+Mentioning a license is only useful if people know what the license means, so add a link to the full text for those who need it.
+
+**Recommended:** No
+
+**Good Example**
+
+```yaml
+asyncapi: "2.0.0"
+info:
+  license:
+    name: MIT
+    url: https://www.tldrlegal.com/l/mit
+```
+
+### asyncapi2-info-license
+
+The `info` object should have a `license` key.
+
+It can be hard to pick a license, so if you don't have a lawyer around you can use [TLDRLegal](https://tldrlegal.com/) and [Choose a License](https://choosealicense.com/) to help give you an idea.
+
+How useful this is in court is not entirely known, but having a license is better than not having a license.
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+asyncapi: "2.0.0"
+info:
+  license:
+    name: MIT
+```
+
+### asyncapi2-operation-description
+
+Operation objects should have a description.
+
+**Recommended:** Yes
+
+### asyncapi2-operation-operationId
+
+This operation ID is essentially a reference for the operation. Tools may use it for defining function names, class method names, and even URL hashes in documentation systems.
+
+**Recommended:** Yes
+
+### asyncapi2-parameter-description
+
+Parameter objects should have a `description`.
+
+**Recommended:** No
+
+### asyncapi2-payload-default
+
+`default` objects should be valid against the `payload` they decorate.
+
+**Recommended:** Yes
+
+### asyncapi2-payload-examples
+
+Values of the `examples` array should be valid against the `payload` they decorate.
```

Comment:
examples here and the default rule above too please, its super valuable and hard to explain.

Reply:
@philsturgeon Fixed

Reply:
@philsturgeon Could you please double check the proposed examples are clear enough?